### PR TITLE
CVSL-2145 update hdc curfewTimes object and types to match api

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/routes/viewingLicences/handlers/viewLicence.ts [hunk 1]
```diff
@@ -61,8 +61,10 @@ export default class ViewAndPrintLicenceRoutes {
         )
       }
 
+      const additionalLicenceConditions = licence.additionalLicenceConditions as AdditionalCondition[]
```
Comment (louisenorris) [inline server/routes/viewingLicences/handlers/viewLicence.ts:64]: Casting the type here to fix the typing issue I was having on ln 67 below. Was erroring with `Argument of type 'string' is not assignable to parameter of type 'never'.`

Comment (louisenorris) [reply server/routes/viewingLicences/handlers/viewLicence.ts]: Fix no longer needed

File: server/routes/approvingLicences/handlers/approvalView.ts [hunk 2]
```diff
@@ -31,8 +32,10 @@ export default class ApprovalViewRoutes {
 
       const returnPath = encodeURIComponent(`/licence/approve/id/${licence.id}/view`)
 
+      const additionalLicenceConditions = licence.additionalLicenceConditions as AdditionalCondition[]
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/approvingLicences/handlers/approvalView.ts]: Fix no longer needed

File: server/data/licenceApiClient.ts [hunk 3]
```diff
@@ -685,9 +685,9 @@ export default class LicenceApiClient extends RestClient {
     })) as Promise<ComCase[]>
   }
 
-  async getHdcLicenceData(bookingId: number): Promise<HdcLicenceData> {
+  async getHdcLicenceData(licenceId: number): Promise<HdcLicenceData> {
     return (await this.get({
-      path: `/hdc/curfew/bookingId/${bookingId}`,
+      path: `/hdc/curfew/licenceId/${licenceId}`,
```
Comment (louisenorris) [inline server/data/licenceApiClient.ts:690]: fixed to match API

File: server/routes/viewingLicences/handlers/printLicence.test.ts [hunk 4]
```diff
@@ -19,6 +20,70 @@ describe('Route - print a licence', () => {
...
+      {
+        curfewTimesSequence: 7,
+        fromDay: 'MONDAY',
+        fromTime: { hour: 17, minute: 0, second: 0, nano: 0 },
+        untilDay: 'SUNDAY',
+        untilTime: { hour: 9, minute: 0, second: 0, nano: 0 },
+      },
+    ],
+  } as HdcLicenceData
+
```
Comment (louisenorris) [inline server/routes/viewingLicences/handlers/printLicence.test.ts:86]: updated to match curfewTimes data in API

File: tsconfig.json [hunk 5]
```diff
@@ -7,7 +7,6 @@
     "moduleResolution": "node",
     "outDir": "./dist",
     "sourceMap": true,
-    "suppressImplicitAnyIndexErrors": true,
```
Comment (louisenorris) [inline tsconfig.json:10]: deprecated so removed

File: server/views/pages/licence/HDC_AP.test.ts [hunk 6]
```diff
@@ -55,33 +55,65 @@ describe('Print a HDC AP licence', () => {
...
 
     expect($('title').text()).toContain('John Smith')
 
-    expect($('#details').text()).toContain('0114 2345232334')
-
-    expect($('#appointmentAddress').text()).toContain('The Square')
+    expect($('#meeting-details').text()).toContain('The Square')
 
-    expect($('#curfewAddress').text()).toContain('addressLineOne')
+    expect($('#curfew-address').text()).toContain('addressLineOne')
```
Comment (andrewrlee) [inline server/views/pages/licence/HDC_AP.test.ts:116]: guess it'd be nice to include a test to show the times coming through?

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**